### PR TITLE
fix: use forked metrics and forked sentry to fix RUSTSEC-2020-0041 temporarily

### PR DESCRIPTION
**Pull request by yangby-cryptape**
### Commits

- [chore(deps): bump crossbeam-channel from 0.4.3 to 0.4.4](https://github.com/nervosnetwork/ckb/commit/728f5c3)
- [chore: remove direct dependencies of metrics exporters and observers](https://github.com/nervosnetwork/ckb/commit/de3173a)
- [chore(deps): bump sentry from 0.16.0 to 0.17.0](https://github.com/nervosnetwork/ckb/commit/1453b1a)
- [fix: use patched versions of metrics and sentry to fix RUSTSEC-2020-0041 temporarily](https://github.com/nervosnetwork/ckb/commit/6aaae31)
  - [Changes of forked `metrics`](https://github.com/nervosnetwork/metrics-rs/commit/4fd13f6)
  - [Changes of forked `sentry`](https://github.com/nervosnetwork/sentry-rust/commit/0e964f5)

### References

- [RUSTSEC-2020-0041](../../../../RustSec/advisory-db/pull/381)

### CODE REVIEW

File: util/app-config/src/sentry_config.rs [hunk 1]
```diff
@@ -31,8 +27,6 @@ impl SentryConfig {
                     scope.set_extra("org_contact", org_contact.clone().into());
                 }
             });
-
-            register_panic_handler();
```
Comment (doitian) [inline util/app-config/src/sentry_config.rs:35]: How the sentry panic handler is registered in the new version?

Comment (yangby-cryptape) [reply util/app-config/src/sentry_config.rs:35]: [`Sentry` will set itself as the panic handler automatically on setup, so no need to do that manually](../../../../getsentry/sentry-rust/issues/235#issuecomment-654832365)

Comment (doitian) [reply util/app-config/src/sentry_config.rs:35]: OK. I'm verifying it.

Comment (doitian) [reply util/app-config/src/sentry_config.rs:35]: http://sentry.nervos.org/share/issue/e7395d8f2f964df3b5d3ef2139626680/

**Review by doitian — DISMISSED** [review]

**Review by driftluo — APPROVED** [review]

**Comment by yangby-cryptape** [comment]
bors merge

**Comment by yangby-cryptape** [comment]
bors retry

**Comment by yangby-cryptape** [comment]
The last force-pushed replace `bump sentry from 0.16.0 to 0.19.1` by `bump sentry from 0.16.0 to 0.17.0`.
Since `ckb` with `sentry >=0.18.0,<=0.19.1` could pass integration tests on MacOS in Travis CI.

I don't have Mac, so it's too hard to debug it for me via Travis CI. **I gave up!**
Maybe someone who has Mac could continue debugging to figure out the reason.

**Review by driftluo — APPROVED** [review]

**Review by doitian — APPROVED** [review]

**Comment by doitian** [comment]
bors retry

**Pull request merged**